### DISCUSSION TITLE
Using Bisq for Peer to Peer BTC loans and interest accounts

### DISCUSSION THREAD

**Discussion by pazza83** [bisq-network/bisq#5213] [Ideas]
Using your crypto to take out loans and access lucrative interest rates is increasing in popularity. 

I thought I would add my thoughts here about how this could be achieved in Bisq for discussion. 

Here is a rough idea of how it could work. I have not given it much thought just an idea that popped up.

**Loans**

- Bob has 2 BTC and wants a loan of $50,000 he would be happy to pay 10% APR for 12 months
- He puts up a maker offer on Bisq
- Alice has $50,000 and would be happy to lend it to Bob for 10% APR. She takes Bob's offer.
- Trade takes place, Bob's 2 BTC enter the multi-sig
- Alice gets confirmation of deposit, she sends Bob the $50,000
- Bob uses the $50,000 to pay the deposit for a new house
- Bob pays back Alice $55,000
- Alice confirms payment receipt Bob gets his funds back

**Interest**

- Bob has 2 BTC and wants to earn some interest 7% APR for 12 months 
- He puts up a maker offer on Bisq for someone that would like to pay him interest
- Alice has 100,000 BSQ and would be happy to pay Bob 7% interest on his BTC. She takes Bob's offer.
- Trade takes place, Bob's 2 BTC enter the multi-sig
- Alice burns 100,000 BSQ 
- Trade is complete and Bob releases Alice the 2 BTC
- Alice uses the 2 BTC to make some lots of trades on Bisq she is making a killing trading BTC for XMR 
- Alice has made lots of profit. At the end of the year she sends Bob back 2 BTC + 7% interest to the address pre stipulated in the terms.
- Alice provides proof to the DAO of the repayment and submits request for 100,000 BSQ to be issued to her.
- DAO confirms repayment and issues Alice with 100,000 BSQ

The numbers above are not based on anything. I did not work out loan-to-value for BSQ / BTC etc. Just used even numbers to keep things simple.

These are just some initial thoughts about how BTC/BSQ could be used as collateral for loans but thought I would share to prompt any ideas / discussion.

**Comment by ghost** [comment]
Borrowing against, instead of selling one's bitcoin is a hot topic right now.  It can be tax-beneficial to take a loan against your assets instead of selling it and paying ~30% CG tax.  Ideas and experiences (including potential downsides) are covered well in https://stephanlivera.com/episode/251/ by Dan Held.

Technical implementation was discussed in https://github.com/bisq-network/proposals/issues/176, so perhaps now with sufficient interest in that feature it could be done?  Ping @stejbac @sqrrm 

[NYKNYC](https://www.youtube.com/watch?v=dnC5mFaIW3Q) applies but I'm more likely to trust the Bisq DAO than the likes of Celcius, Nexo, BlockFi, [Unchained Capital](https://unchained-capital.com/loans/).

--- 

[edit] https://github.com/bisq-network/proposals/issues/176 is for BSQ/BTC loans.  @pazza83 idea is for Fiat loans, both could be useful.  More use-case examples would be good, esp the non-happy path scenarios.

**Comment by Conza88** [comment]
I would be inclined to say I don't think we've got trading sorted to an acceptable level (i.e. lack of multiple payment methods per offer which increases liquidity immensely for everyone), so this should be ball packed until that is clearly the case. [https://github.com/bisq-network/proposals/issues/288](https://github.com/bisq-network/proposals/issues/288)

Both use cases are positive. Would be interested in negative scenarios of being unable to pay back etc.

**Comment by pazza83** [comment]
@jmacxx 

>Technical implementation was discussed in bisq-network/proposals#176, so perhaps now with sufficient interest in that feature it could be done? Ping @stejbac @sqrrm

Thanks I did not see that before. I will take a look.

> [edit] bisq-network/proposals#176 is for BSQ/BTC loans. @pazza83 idea is for Fiat loans, both could be useful. More use-case examples would be good, esp the non-happy path scenarios.

@Conza88 

> I would be inclined to say I don't think we've got trading sorted to an acceptable level (i.e. lack of multiple payment methods per offer which increases liquidity immensely for everyone), so this should be ball packed until that is clearly the case. bisq-network/proposals#288

I agree, just thinking about potential use cases for future projects. 


### Trades gone bad

**Loans**

Here is the bad case for loans. In this case I assume the Bisq user that lends $50,000 is going to prefer BTC over fiat so their upside if things go wrong is the BTC collateral.

- Bob has 2 BTC and wants a loan of $50,000 he would be happy to pay 10% APR for 12 months
- He puts up a maker offer on Bisq
- Alice has $50,000 and would be happy to lend it to Bob for 10% APR. She takes Bob's offer.
- Trade takes place, Bob's 2 BTC enter the multi-sig
- Alice gets confirmation of deposit, she sends Bob the $50,000
- Bob uses the $50,000 to pay the deposit for a new house
- Bob cannot pay back Alice $55,000
- Alice opens mediation
- Mediators award Alice 90% of what is in the multi-sig and award Bob 10%. If partial payment received this would be awarded as a percentage of the partial payment.



**Interest**

Here is the bad case for interest. In this case I assume 1 BTC = 25,000 BSQ

- Bob has 2 BTC and wants to earn some interest 7% APR for 12 months 
- He puts up a maker offer on Bisq for someone that would like to pay him interest
- Alice has 100,000 BSQ and would be happy to pay Bob 7% interest on his BTC. She takes Bob's offer.
- Trade takes place, Bob's 2 BTC enter the multi-sig
- Alice burns 100,000 BSQ 
- Trade is complete and Bob releases Alice the 2 BTC
- Alice uses the 2 BTC to make some lots of trades on Bisq she is loses all her funds with some poor trades. 
- Alice has lost everything. At the end of the year she cannot send Bob back 2 BTC + 7% interest to the address pre stipulated in the terms.
- Bob provides proof to the DAO that the repayment did not occur and requests Alice's 100,000 BSQ to be issued to him.
- DAO confirms no repayment and issues Bob with 100,000 BSQ

**Comment by chimp1984** [comment]
A simple implementation would be a trade with a very long trade duration (the repayment period). The fiat need to be sent asap but the trade does not get completed but instead the BTC seller sends back the fiat after the agreed period. Then it would be nice to have a trade complete path where the seller gets back the BTC (like for the cancel trade feature), otherwise they need to open mediation to close it in favor of the seller.
In that model all security is already baked in, as a non-refund of the fiat would lead to a normal trade where the seller has to release the BTC for the received fiat.

**Reply by pazza83** [reply]
Thanks for the feedback. Does that mean it would be not require much dev resources to implement?

**Reply by chimp1984** [reply]
I guess conceptually there is not missing much (it would work already without any change if both traders are aware of the "deal"). To get that into a nice easy to understand UI is another challenge.
Whats needed as a minimum for UX is:
- A refund button to send do the payout of the trade amount to the seller would be good, otherwise the mediators would need to play that role.
- The trade time need to be extended (could be done via a custom payment method).

The Bisq trade protocol is an option trade as there is significant time delay and the security deposit is the price for the option. Lending is a future contract where the lender has both options: To pay back the load or to default and pay the price for the option via lost security deposit.

**Comment by RififiCastorjunior** [comment]
Bisq is willing to Guaranty all terms, of all contracts for the period of those contract times, meaning if something go wrong Bisq will take responsibility to reimburse all.
 
At the moment Bisq doesnt stand in those loans and INTERESTS schemes meaning if Bisq accept anything relate to interest there some people that won't come to the plateform. Religious peoples goes to paxful because of this thing only, not because they have outrage fees but because of this mechanism of business. 

In this exemple i just take Muslim people not other religious country. 

![Muslim_majority_countries2](https://user-images.githubusercontent.com/67669956/110440041-81813b00-80b8-11eb-9d38-ea094e36b261.png)
![Muslim-flags-Countries](https://user-images.githubusercontent.com/67669956/110440053-8645ef00-80b8-11eb-8ee5-d5e5b830e158.jpg)
Fig1&2 are the country Bisq will have to forget plus others relate to peoples believes.

**Reply by pazza83** [reply]
Hi @RififiCastorjunior 

Thanks for the information. I must admit I do have any real insight into Islamic finance. I am assuming that is what you are referring to?

Please can you explain what Plaxful are doing that make their platform suitable for people wanting to adhere to Islamic finance beliefs.

**Reply by RififiCastorjunior** [reply]
Hi! @pazza83 Yes, that the rule of "ribâ" aka interest, i think its the only big major rule in islamic finance other religious people also follow that rule but they are not strict as is it for muslim people. 

It's a pretty simple rule. it can summarize it as follows : 
 - no interest of any kind on loans.
 - if loan is taken they should be no expire date to repay the loan.

Those are the simple way to explain this rule. But this rule goes beyond and explain all kind of situation and have scholar on this matter. 
Paxful doesn't use loan and interest and will never add that to his platform. They make sûre people have P2p exchange and that all. And because is a website it can be use by all kind of bankless all over the globe. When they figure out how to lower their fees or integreate a L2 solution their will be a major exchange for all bankless in crypto.

**Reply by Conza88** [reply]
In Islamic translation just call it a "fee".
Seriously. 0% loan. Just with a fee for whatever.

Zippay in Australia, market 0% interest free loans. There is an admin fee of $5 per month balance is outstanding... 🤣

That is really just another way of charging the interest.

Problem solved.

**Reply by RififiCastorjunior** [reply]
There are a lot of offers that call themselves "legal" aka Halal but muslim people still studies and find those laughable and tricky offers , they also make sure to remember for all of their communities. I think zippay is no adopt by those religious people in autralia.


This rules go beyond those two point. They have history of people fool them so they make sure to remember those schemes.
The boycott for palestinian is a good example. Company are mark and avoid doing business with them.


For them to take those "interest" aka ribâ is the same as to sleep with their mother (*′☉.̫☉) that is written in the Quran as example.
So they take this rule **really seriously**. 


There are not a lot of places that don't use loans and interest in crypto so if Bisq keep up without those two. It easily marketable for all Religious people. Count all "people of the book" on this matter.

**Reply by AzidanoValkran** [reply]
Sorry for bringing this thread back from the dead. I just wanted to give my 2 cents on the mention of riba and Islam.

In classical Islam there is quite a bit of evidence to say that riba (unlawful surpluss, colloquially equivilant to interest) is prohibited, but in terms of its adherence amongst Muslims is predictably varied. I have no measurables or data to support this, so please bear that in mind:

There is a small inner circle of adherents which will have nothing to do with interest, not only does this include not participating in the Bisq network should it provide interest-based facilities, but also would not open bank accounts, would not work for companies that deal in interest, would not get credit cards (even if they pay them off one time and get charged no interest), not even have fiat which is associated with bonds thus interest. Bisq is probably the least impactful on their conduct as they would, effectively wish to divorce themselves from the entire financial system.

Next out people become more and more relaxed, making more and more compromises.

The point I am trying to raise is that anybody who would refuse to use Bisq because it offers an interest-based faciliation, would also not have bank accounts.

Most Muslims in most countries have bank accounts, but would not take out any loans. Similarly, my guess is that these very people would happily use Bisq for trading Bitcoin even if they would avoid the interest part of it.

I think a greater concern is that there are many Muslim scholars who are actually opining that Bitcoin and cryptocurrencies are Haram, anyway.

**Comment by AzidanoValkran** [comment]
I am a regular user of Hodl Hodl, which appears to have a similar philosophy to Bisq (whereas Bisq, in my opinion, is a truly decentralised implementation) But this site also offers a faciliation service for loans where they use stablecoins. My guess is that if you stick to stablecoins it is much easier to verify independently if claimed payments were actually made or not.

I have a few questions for @pazza83, regarding his idea:

1. How would Bisq know when a loan has defaulted (would this be the same mechanism as the payment window?)
2. Would these be mark-to-market loans? (If they are this creates a load of new questions)
3. For each loan offer the does person decide how much BTC they want to ask for and how much they want to loan?

**Reply by pazza83** [reply]
Thanks for the message.

It is a while since I posted this but will try and answer your questions.

1. Bisq would know the loan has defaulted as the lender would open mediation. It could be the same process as is done know for BTC buyers that do not send fiat payment. Yes payment window but be the same as present just longer for the duration of the loan.
2. I do not understand this question. I would anticipate the loan being over collateralized at the time of the offer being taken and that would be marked to market at this time. 
3. Good question. I assume the offer maker would decide how much BTC they would like and this would then be over collateralized by an amount in BSQ calculated from the current price.

**Reply by AzidanoValkran** [reply]
The answer to the first question is a good one. The mediation mechanism helps a lot here.

To clarify my second question, as far as I know, mark-to-market loans are loans where if the collateral dips below a certain value, a margin call is sent, if it dips even lower than this then the loan is liquidated, meaning that the collateral is immediately sent to the lender as a means of settling the loan. This method raises further questions such as what is used as the oracle for prices and how often are these checked to trigger liquidation.

For the third question. My suggestion would be to set the percentage of the Loan-to-Value in the offer in a similar way that the price is set ? Mixing collateral types doesn't seem like it would work.

**Comment by pazza83** [comment]
> To clarify my second question, as far as I know, mark-to-market loans are loans where if the collateral dips below a certain value, a margin call is sent, if it dips even lower than this then the loan is liquidated, meaning that the collateral is immediately sent to the lender as a means of settling the loan. This method raises further questions such as what is used as the oracle for prices and how often are these checked to trigger liquidation.

For my proposal I did not anticipate loans being liquidated. There would be no calling in the loan early or liquidations. Other platforms cannot really do this as they are centralized and it would be too high a risk. Bisq could do it because loans are p2p and the risk falls onto the individual users of the platform. For that reason interest would be accordingly priced and would likely be higher than interest charged in a situation where ones collateral can be liquidated to cover the loan.

> For the third question. My suggestion would be to set the percentage of the Loan-to-Value in the offer in a similar way that the price is set ? Mixing collateral types doesn't seem like it would work.

Can you provide an example?

**Reply by AzidanoValkran** [reply]
> Can you provide an example?

Yeah sure. So I am currently trying to lift and shift the way Hodl Hodl do it, and see if it could be applied in a truly decentralised p2p environment such as Bisq. Let us say that, for this example the current price of BTC is $10,000 per coin.

I have $10,000 that's like to lend out at 10% APR for one year, and the LTV ratio I am prepared to offer is 50%. This means that my loan should be worth 50% of the current market value (meaning the collateral should be worth double)

Thus my Offer would look something like this:

| Loan Amount | Amount to Receive | Loan Duration | Collateral Requested| Calculated APR | Calculated LTV |
|--------|--------|--------|--------|--------|--------|
| $10,000 | $11,000 | 1 Year | **_2 BTC_** | 10% | 50% |

The final columna of APR and LTV is just to allow Users to sort by those columns as they would seem to be of interest to potential lenders.

When the price of Bitcoin changes, my suggestion is to keep the loan amounts and percentages the same but change the amount of BTC required. Let us say the price changes from $10,000 to $20,000:

| Loan Amount | Amount to Receive | Loan Duration | Collateral Requested| Calculated APR | Calculated LTV |
|--------|--------|--------|--------|--------|--------|
| $10,000 | $11,000 | 1 Year | **_1 BTC_** | 10% | 50% |

**Reply by pazza83** [reply]
Ok so in Bisq the BTC is fixed so when price changes the loan amount offered would go up

So if the price of BTC increased from 10k to 20k would look something like this:

| Loan Amount | Amount to Receive | Loan Duration | Collateral Requested| Calculated APR | Calculated LTV |
|--------|--------|--------|--------|--------|--------|
| $20,000 | $22,000 | 1 Year | **_2 BTC_** | 10% | 50% |

**Reply by pazza83** [reply]
So the lender would not define a dollar amount of loan they were happy to make. They would define the amount of BTC they want as collateral and the dollar amount would be calculated from this and changed with the market until the offer was taken.

**Reply by AzidanoValkran** [reply]
Ahh yes, I had forgotten about that. It would be similar to the way somebody puts an offer on Bisq to buy some BTC. If the price goes up 10x, then they'll be obligated to pay 10x the prices, assuming they put on a relative price.

One of the things that is most attractive about Bisq, is the requirement for a deposit, which serves as a filtering mechanism so you only get genuine interest (or at least interest who are will to put skin in the game, so to speak)

I wonder how such a mechanism could work for this?

**Comment by pazza83** [comment]
Security deposit would work in the same way as present.

That way if either the lender or borrow do no go though with the loan then they would lose some of their 15% deposit user for security.

Bob wants to borrow $10,000 and takes Alice's offer
Alice goes AWOL does not send Bob the $10,000
Bob opens mediation
Mediator proposes loan to be cancelled and Bob to be compensated with 15% of Alice's security deposit.

or from the other perspective 

Bob wants to borrow $10,000 and takes Alice's offer
Alice pays back Bob the $10,000 + $1,000 interest
Bob goes AWOL and does not release Alice's collateral 
Alice opens mediation
Mediator proposes loan to be finalized and Alice to be compensated with 15% of Bobs security deposit, plus the return of her collateral

**Reply by AzidanoValkran** [reply]
This makes a lot of sense. I feel the second scenario would play a bit more like this:

Bob wants to borrow $10,000 and takes Alice's offer
Alice sends $10,000 to Bob
A year later Bob sends Bob $10,000 + $1000 interest
Bob goes AWOL and does not release Alice's collateral
Alice opens mediation
Mediator proposes loan to be settled by releasing collateral to Alice and for her to be compensated with 15% of Bob's security deposit

There is also another potential failure point here:
Bob wants to borrow $10,000 and takes Alice's offer
Alice sends $10,000 to Bob
A year later Bob goes AWOL and does not repay anything back to Alice
Alice opens mediation
Mediator proposes loan to be settled by releasing collateral to Alice and for her to be compensated with 15% of Bob's security deposit.

I really like this idea!

**Reply by pazza83** [reply]
Yes, this is where the security deposits come in useful :)

I think more innovative things can be developed in a trustless way using bitcoin secured in a multisig. 

I think something like this is a while off being developed for Bisq but would be nice to see included in something like a Bisq 2 trade protocol in the future.
